tools/esp8266/web.cpp

```diff
@@ -563,11 +570,80 @@ void web::showLiveData(void) {
 
 
 //-----------------------------------------------------------------------------
+#ifdef ENABLE_JSON_EP
 void web::showJson(void) {
     DPRINTLN(DBG_VERBOSE, F("web::showJson"));
-    mWeb->send(200, F("application/json"), mMain->getJson());
+
+    String modJson;
+
+    modJson = F("{\n");
+    for(uint8_t id = 0; id < mMain->mSys->getNumInverters(); id++) {
+        Inverter<> *iv = mMain->mSys->getInverterByPos(id);
+        if(NULL != iv) {
+            char topic[40], val[25];
+            snprintf(topic, 30, "\"%s\": {\n", iv->name);
+            modJson += String(topic);
+            for(uint8_t i = 0; i < iv->listLen; i++) {
+                snprintf(topic, 40, "\t\"ch%d/%s\"", iv->assign[i].ch, iv->getFieldName(i));
+                snprintf(val, 25, "[%.3f, \"%s\"]", iv->getValue(i), iv->getUnit(i));
+                modJson += String(topic) + ": " + String(val) + F(",\n");
+            }
+            modJson += F("\t\"last_msg\": \"") + mMain->getDateTimeStr(iv->ts) + F("\"\n\t},\n");
+        }
+    }
+    modJson += F("\"json_ts\": \"") + String(mMain->getDateTimeStr(mMain->mTimestamp)) + F("\"\n}\n");
+
+    mWeb->send(200, F("application/json"), modJson);
+}
+#endif
+
+//-----------------------------------------------------------------------------
+#ifdef ENABLE_PROMETHEUS_EP
+
+std::pair<String, String> web::convertToPromUnits(String shortUnit) {
+
+    if(shortUnit == "A")    return {"amplere", "gauge"};
+    if(shortUnit == "V")    return {"volts", "gauge"};
+    if(shortUnit == "%")    return {"ratio", "gauge"};
+    if(shortUnit == "W")    return {"watts", "gauge"};
+    if(shortUnit == "Wh")   return {"watts_daily", "counter"};
+    if(shortUnit == "kWh")  return {"watts_total", "counter"};
+    if(shortUnit == "°C")   return {"celsius", "gauge"};
```

Comment:
amplere should be ampere.
for volts and watts you use the plural form, volt and watt would be the common unit ?
°C should be degree_celsius then ?

Reply:
corrected the units using the singular version. I left the temperature `celsius` since I like to keep them as short as possible